### PR TITLE
fix sync issue after block 6800 and specialfee math

### PR DESCRIPTION
**Pull request by npq7721**

### CODE REVIEW

File: src/miner.cpp [hunk 1]
```diff
@@ -195,10 +195,11 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     coinbaseTx.vout[0].scriptPubKey = scriptPubKeyIn;
 
     // NOTE: unlike in bitcoin, we need to pass PREVIOUS block height here
-    CAmount blockReward = nFees + nSpecialTxFees + GetBlockSubsidy(pindexPrev->nBits, pindexPrev->nHeight, Params().GetConsensus());
+    CAmount normalBlockReward = nFees + GetBlockSubsidy(pindexPrev->nBits, pindexPrev->nHeight, Params().GetConsensus());
+    CAmount blockRewardWitSpecialtx = normalBlockReward + nSpecialTxFees;
```
Comment (CodeIsTheKey) [inline src/miner.cpp]: Typo
```suggestion
    CAmount blockRewardWithSpecialtx = normalBlockReward + nSpecialTxFees;
```

File: src/miner.cpp [hunk 2]
```diff
@@ -195,10 +195,11 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
...
 
     // Compute regular coinbase transaction.
-    coinbaseTx.vout[0].nValue = blockReward;
+    coinbaseTx.vout[0].nValue = blockRewardWitSpecialtx;
```
Comment (CodeIsTheKey) [inline src/miner.cpp]:
```suggestion
    coinbaseTx.vout[0].nValue = blockRewardWithSpecialtx;
```

**Review by CodeIsTheKey — APPROVED** [review]
Looks good to me - glad we got past 6800!

File: src/miner.cpp (hunk 1 above)
Comment (npq7721) [reply src/miner.cpp]: fixed

**Pull request merged**